addons/enemybar/actionTracking.lua

```diff
@@ -0,0 +1,299 @@
+					tracked_enmity[mob.id] = {pc=pc, mob=mob, time=os.time()}
+
+					-- if the actor is the npc, we don't care about the other targets. The first target is the one they targeted.
+					if actor == mob then return end
+				end
+			end
+		end
+	end
+end
+
+function track_actions(ai)
+	local actor = windower.ffxi.get_mob_by_id(ai.actor_id)
+	if not actor then return end
+
+	local target = windower.ffxi.get_mob_by_id(ai.targets[1].id)
+	if not target then
+		target = {name='Unknown (id:'..ai.targets[1].id, id=ai.targets[1].id}
+	end
+
+	-- if the category is not casting magic, jas, items or ws, don't bother.
+	if not S{8,4,7,11,3,6,9,5}:contains(ai.category) then return end
+
+	-- if it's a starting packet, the id is in param2
+	local action_id = ai.param
+	if S{8,7,9}:contains(ai.category) then
+		action_id = ai.targets[1].actions[1].param
+	end
+
+	-- find the action
+	local action_map = nil
+	if S{8,4}:contains(ai.category) then
+		action_map = res.spells[action_id]
+	elseif S{5,9}:contains(ai.category) then
+		action_map = res.items[action_id]
+	else
+		if actor.is_npc then
+			action_map = res.monster_abilities[action_id]
+		else
+			if ai.category == 6 then
+				action_map = res.job_abilities[action_id]
+			elseif S{3,7,11}:contains(ai.category) then
+				action_map = res.weapon_skills[action_id]
+			end
+		end
+	end
+	-- couldn't find the action, let's just give some debug output.
+	if not action_map then
+		action_map = {en='Unknown (id:'..action_id..')'}
+	end 
+
+	tracked_actions[ai.actor_id] = {actor=actor, target=target, ability=action_map, complete=S{4,11,3,5,6}:contains(ai.category), time=os.time()}
+end
+
+function track_debuffs(ai)
+	check_conflicting_debuffs(ai)
+	if S{8,7,6,9}:contains(ai.category) then return end
+
+	for i,t in ipairs(ai.targets) do
+		local target = windower.ffxi.get_mob_by_id(t.id)
+		if target and target.is_npc and not target.charmed then
+
+		    if S{2,252}:contains(t.actions[1].message) then
+		        local spell = ai.param
+		        local effect = res.spells[spell].status
+
+		        if effect then
+		            apply_debuff(target, effect, spell)
+		        end
+		        
+		    -- Non-damaging spells
+		    elseif S{75,236,237,268,270,271}:contains(t.actions[1].message) then
+		        local effect = t.actions[1].param
+		        local spell = ai.param
+
+		        apply_debuff(target, effect, spell)
+		    end
+		end
+	end
+end
+
+function apply_debuff(target, effect, spell)	
+    if not tracked_debuff[target.id] then
+        tracked_debuff[target.id] = {}
+    end
+
+    local overwrites = res.spells[spell].overwrites or {}
+    if not did_overwrite(target, spell, overwrites) then
+        return
+    end
+
+    tracked_debuff[target.id][effect] = {target=target,spell=spell,effect=effect,time=os.time(),duration=res.spells[spell].duration or 0,pos={x=target.x,y=target.y}}
+end
+
+function did_overwrite(target, new, t)
+    if not tracked_debuff[target.id] then
```

Comment:
This check can never be `true`. You check for that already in the calling function, and if `true` you set an empty table as the value, which will make this check always evaluate to `false`.